x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Page.java

```diff
@@ -222,6 +221,43 @@ public void writeTo(StreamOutput out) throws IOException {
      */
     public void releaseBlocks() {
         blocksReleased = true;
-        Releasables.closeExpectNoException(blocks);
+        // blocks can be used as multiple columns
+        var map = new IdentityHashMap<Block, Object>(mapSize(blocks.length));
+        var DUMMY = new Object();
```

Comment:
That could be moved into a static final var.

Reply:
Or...

```
var map = new IdentityHashMap<Block, Boolean>(mapSize(blocks.length));
for (Block b : blocks) {
    if (map.putIfAbsent(b, Boolean.TRUE) == null) { ..
```